src/experimental/tools/videoThumbnailLoader/thumbnail_loader.ts

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import pinkie from "pinkie";
+import {
+  race as observableRace,
+  Subject,
+} from "rxjs";
+import {
+  catchError,
+  filter,
+  map,
+  mergeMap,
+  take,
+} from "rxjs/operators";
+import createSegmentLoader from "../../../core/pipelines/segment/create_segment_loader";
+import Player from "../../../index";
+import log from "../../../log";
+import { ISegment } from "../../../manifest";
+import dash from "../../../transports/dash";
+import getContentInfos from "./get_content_infos";
+import {
+  disposeSourceBuffer,
+  initSourceBuffer$,
+} from "./init_source_buffer";
+import removeBuffer$ from "./remove_buffer";
+import { IContentInfos } from "./types";
+import VideoThumbnailLoaderError from "./video_thumbnail_loader_error";
+
+const PPromise = typeof Promise === "function" ? Promise :
+                                                 pinkie;
+
+interface IJob { contentInfos: IContentInfos;
+                 segment: ISegment;
+                 stop: () => void;
+                 jobPromise: Promise<unknown>; }
+
+const segmentLoader = createSegmentLoader(
+  dash({ lowLatencyMode: false }).video.loader,
```

Comment:
So as I wrote earlier for the init segment:
  1. we must think about how we will adapt it once hls comes
  2. why using only the loader here and not both the loader and parser (or better yet, the fetchers!)